### ISSUE TITLE
Please come back to YCSB

### ISSUE THREAD

**Issue by busbey** [YCSB/YCSB#5]
Hi Folks!

I understand the frustration with YCSB's long hiatus. I certainly had my share of shouting into the void over the period of inactivity.

However, @brianfrankcooper has recognized the problem and is working proactively to get the community back on track. We're attempting to build a new set of maintainers to ensure we have robust activity going forward. Brian's already started by enlisting @cmccoy and me; we have a good start at refocusing things for a community effort.

We'd love it if you would come work with us rather than duplicating effort on a divergent fork.

**Comment by nitsanw** [comment]
This is a new effort, and as of now it's hard to say which way things would go. Certainly if it turns out that the @brianfrankcooper repo is being maintained and PRs are merged in and so on it would be great. The problems we set out to solve with this fork were:
1. YCSB original repository was stale, unmaintained and unresponsive
2. Original repository was affiliated with a particular person, and his employment at Yahoo.
I have no doubt BFG is an honest and impartial person, but I think his affiliation with a particular employer makes his ownership problematic.
To resolve this, and make YCSB a communal effort I setup this repo under an organization which I intend to keep open to maintainers from all vendors. I think this is the right structure for an industry and community standard benchmark.
I would be happy to have you join this effort, along with BFG and @cmccoy and any vendor representatives, I think at this point in time the differences are still easy enough to bridge. Alternatively you may choose to move YCSB into a more communal ownership model, or convince involved parties this is really not an issue...
A further thought on this. I believe part of the difficulty in YCSB's maintenance  is because of it's monolithic project structure mixing the core measurement part with vendor code. A looser model would minimize the need for vendor merges and reduce the overhead of refining the benchmark harness.
In any case I'm very happy to hear the project is coming back to life, I look forward to our collaboration in whichever flavour it may come.

**Comment by daschl** [comment]
Yeah, of course I'd also appreciate a single source of truth instead of yet another fork. 

@jbellis ?

**Comment by jbellis** [comment]
I think the world has made its peace with bfc/YCSB bitrotting, so I think in a lot of ways it would be easier for interested parties to join our effort here rather than try to put the horse back in the barn.  It's also a better code base to work with than the mostly-2013-era bfc repo.  Please give it some thought @busbey and @cmccoy.

**Comment by cmccoy** [comment]
Thanks for responding, all.

I certainly think there's a lot of value in having a single source of truth.  
I prefer maintaining brianfrankcooper/YCSB as the canonical repo: @busbey has been doing a lot of work to bring it back up to speed, and we're interested in adding additional maintainers (including you, if interested!) so that the project doesn't go stagnant again. If there are enhancements in this codebase, they can be merged in.

**Comment by jbellis** [comment]
Shouldn't step zero be, look at the different forks and decide which one to build off of (whether in the forked repo or by pulling it in to a new bfc branch) before charging ahead with commits to the very old bfc-classic branch?

**Comment by busbey** [comment]
> Shouldn't step zero be, look at the different forks and decide which one to build off of (whether in the
> forked repo or by pulling it in to a new bfc branch) before charging ahead with commits to the very old 
> bfc-classic branch?

There are a multitude of forks and no easy way to prioritize them. We decided to go with pulling in changes from all of them as we can. Some folks have made that easier than others, and since reviewer time is our most constrained resource that effectively determines the merge order.

**Comment by jbellis** [comment]
I would actually say (having done it) that it's pretty easy to prioritize them.  The vast majority of them have just a tweak or two, generally to a single vendor's client.  The other one that I saw with substantive work was Thumbtack's, whose changes I pulled in to this one.
